Expose config option to enable low-level S3 HTTP operation logging. (#8690)
grafana/mimir: docs/sources/mimir/configure/configuration-parameters/index.md

  # (advanced) Override the expected name on the server certificate.
  # CLI flag: -<prefix>.s3.http.tls-server-name
  [tls_server_name: <string> | default = ""]

trace:

>   # (advanced) If enabled, low-level S3 HTTP operation info will be logged at
>   # debug level.

Reviewer: ```suggestion
  # (advanced) If enabled, low-level S3 HTTP operation information is logged at the
  # debug level.
```

Reviewer: We try to avoid using "will" in the docs.
